File: backend/models/progress.py
from pydantic import BaseModel


class TopicMastery(BaseModel):
    topic_id: str
    mastery_level: float = 0.0
    badge: str = "none"
    questions_attempted: int = 0
    questions_correct: int = 0
    current_difficulty: int = 1
    next_review_at: str | None = None


class LessonProgressItem(BaseModel):
    topic_id: str
    lesson_id: str
    completed: bool = False
    last_segment_index: int = 0


class ProgressSummary(BaseModel):
    topics: list[TopicMastery]
    lessons: list[LessonProgressItem]
    total_lessons_completed: int = 0
    total_quizzes_taken: int = 0

File: frontend/src/types/progress.ts
export interface TopicMastery {
  topic_id: string;
  mastery_level: number;
  badge: string;
  questions_attempted: number;
  questions_correct: number;
  current_difficulty: number;
  next_review_at: string | null;
}

export interface LessonProgressItem {
  topic_id: string;
  lesson_id: string;
  completed: boolean;
  last_segment_index: number;
}

export interface ProgressSummary {
  topics: TopicMastery[];
  lessons: LessonProgressItem[];
  total_lessons_completed: number;
  total_quizzes_taken: number;
}

export interface CurriculumModule {
  id: string;
  slug: string;
  title: string;
  description?: string;
  path: string;
  mastery?: {
    level: number;
    badge: string;
    questions_attempted: number;
  };
  lessons: {
    id: string;
    title: string;
    path: string;
    completed?: boolean;
  }[];
}

export interface CurriculumTier {
  id: string;
  title: string;
  modules: CurriculumModule[];
}

export interface Curriculum {
  tiers: CurriculumTier[];
}
